tests/test_api.py

```diff
@@ -228,3 +229,65 @@ def test_extract_to_dict(fixtures_path: Path):
         "SEX": {},
         "RACE": {},
     }
+
+
+@pytest.mark.vcr
+def test_submit_extract_live(live_api_client: IpumsApiClient):
+    """
+    Confirm that test extract submits properly
+    """
+    extract = UsaExtract(
+        ["us2012b"],
+        ["AGE", "SEX"],
+    )
+
+    live_api_client.submit_extract(extract)
+    assert live_api_client.extract_status(extract) == "queued"
+
+
+@pytest.mark.vcr
+def test_download_extract(live_api_client: IpumsApiClient):
+    """
+    Confirm that extract data and attendant files can be downloaded
+    """
+    live_api_client.download_extract(collection="usa", extract="136")
+
+
+@pytest.mark.vcr
+def test_download_extract_stata(live_api_client: IpumsApiClient):
+    """
+    Confirm that extract data and attendant files (Stata) can be downloaded
+    """
+    live_api_client.download_extract(
```

Comment:
To avoid actually ending up with the files in your gitignore, I'd suggest the following:

```
import tempfile
from pathfile import Path

@pytest.fixture(session="function")
def tmpdir() -> Path:
    with tempfile.TemporaryDirectory() as tmp:
        yield Path(tmp)

@pytest.mark.vcr
def test_download_extract_stata(live_api_client: IpumsApiClient, tmpdir: Path):
    """ ... """
    extract = ...

    live_api_client.download_extract(
        collection="usa", extract="136", stata_command_file=True, download_dir=tmpdir
    )

    assert (tmpdir / "stata_file.do").exists()

Reply:
Thanks Kevin! The `tempfile` solution feels much more elegant that my `.gitignore` solution. 

One question: Is the `assert [file].exists()` statement necessary? I may have gotten confused trying to move between `vcr.py` and `pytest-recording`, but my understanding of the way I had things set up was that the existence of the .dat, .xml, .do files, etc. was the result of creating and re-playing the `vcr.py` cassettes and that the files themselves were not actually used or accessed in any tests (maybe they should be?). I tested this by deleting the files, switching off my wifi connection and re-running the tests and, as I expected, the files reappeared and all tests passed. My understanding is that even without an assert statement, `vcr.py` is comparing http requests and will fail if the would-be http request does not match its cassette on file. Since the .dat, .do, etc. files (I think) are being generated by replaying the cassette and not actually being downloaded from an actual fresh API response, what is the `.exists()` statement checking? Alternatively, am I misunderstanding `vcr.py` functionality?

Reply:
@khwilson, I am going to go ahead and merge in this PR if everything looks fine to you.